tests/test_snapcraft_store_api.py

```diff
@@ -0,0 +1,138 @@
+from vcr_unittest import VCRTestCase
+from canonicalwebteam.store_api.stores.snapcraft import SnapcraftStoreApi
+
+
+class SnapcraftApiTest(VCRTestCase):
+    def setUp(self):
+        # configure self.attribute
+        self.client = SnapcraftStoreApi()
+        return super(SnapcraftApiTest, self).setUp()
+
+    def test_get_all_items(self):
+        result = self.client.get_all_items(size=16, api_version=1)
+        snaps = result["_embedded"]["clickindex:package"]
+        self.assertEqual(len(snaps), 16)
+
+        # Test different size
+        result = self.client.get_all_items(size=4, api_version=1)
+        snaps = result["_embedded"]["clickindex:package"]
+        self.assertEqual(len(snaps), 4)
+        [self.assertIn("snap_id", snap) for snap in snaps]
+
+    def test_get_category_items(self):
+        result = self.client.get_category_items(
+            category="security", size=10, page=1, api_version=1
+        )
+        snaps = result["_embedded"]["clickindex:package"]
+
+        self.assertEqual(len(snaps), 10)
+
+        for snap in snaps:
+            sections = [i["name"] for i in snap["sections"]]
+            self.assertIn("security", sections)
+
+        # Test different size and category
+        result = self.client.get_category_items(
+            category="server-and-cloud", size=2, page=1, api_version=1
+        )
+        snaps = result["_embedded"]["clickindex:package"]
+
+        self.assertEqual(len(snaps), 2)
+
+        for snap in snaps:
+            sections = [i["name"] for i in snap["sections"]]
+            self.assertIn("server-and-cloud", sections)
+
+    def test_get_featured_items(self):
+        result = self.client.get_featured_items(size=10, page=1, api_version=1)
+        snaps = result["_embedded"]["clickindex:package"]
+        self.assertEqual(len(snaps), 10)
+
+        # Test different size
+        result = self.client.get_featured_items(size=3, page=1, api_version=1)
+        snaps = result["_embedded"]["clickindex:package"]
+        self.assertEqual(len(snaps), 3)
+
+        # Test that snaps are featured
+        for snap in snaps:
+            sections = [i["name"] for i in snap["sections"]]
+            self.assertIn("featured", sections)
+
+    def test_get_publisher_items(self):
+        result = self.client.get_publisher_items(
+            publisher="28zEonXNoBLvIB7xneRbltOsp0Nf7DwS", api_version=1
```

Comment:
What's this hash-like string? Is that an identified for the publisher? Or an actual publisher's name?

If it's a hash of some sort, I think the argument name should be changed to make this clearer. But again, that should be out of scope for this PR, as it makes sense to just focus on testing in this PR.

Reply:
Yes, it is the publisher id

Reply:
Filed: https://github.com/canonical-web-and-design/canonicalwebteam.store-api/issues/12